### PR TITLE
Optimized Istio Config list loading

### PR DESCRIPTION
**Pull request by hhovsepy**
RFE https://github.com/kiali/kiali/issues/5153

Before, while loading Istio Config list:
It was requesting configs per namespace.

Now: it does not provide namespace list into API call, and the handler loads all configs for all namespaces.
This approach is used,  as handler is now loading configs from Registry, where all configs are cached for all namespaces.

### CODE REVIEW

File: frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx
Comment (jshaughn) [reply frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx:156]: Having said that, I'm not sure I understand why we would fetch all namespaces here when the user may often have a small percentage selected.   I can understand for the Overview page, where we usually want them all, but not sure about this location, where the namespace selector is being used.

**Review by jshaughn — COMMENTED** [review]
@hhovsepy, the changes seem to be working,  I did have a suggestion and a question before moving to approval.

File: frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx
Comment (hhovsepy) [reply frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx:156]: The idea of fetching all namespaces comes from the implementation, where the Istio registry is used and all namespaces are read from registry at once in back-end.
Anyway, there will be done filtering of necessary namespaces only. If that filtering will be done in backend, this means that list of namespaces will be sent in http get params, and in case of many namespaces, there will require several requests not to reach the request size limit.
So I decided to load all namespaces at once and then filter in UI.
@jshaughn  wdht?

Comment (jshaughn) [reply frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx]: I think it's OK for now.  I was concerned, for example, about the potential overhead of sending perhaps 100s, or 1000s of Istio config objects over the wire and into the UI, when perhaps only a few were relevant to the Istio Conflig list page.   I was not concerned about the backend, I understand that we are fetching all at the same time, but only the overhead of the response.  But, that can always be optimized later if it is a problem, but sending namespace names in the GET.

Just for my information, because I don't know,  what is the timing of the validations on the istio config?  Is this done at fetch time?  Is it cached?  Do we wait for the first client request to perform validations or do we perform this at startup in order to be "ready in advance"?  And if it is precalculated/cached, when is is updated?  Thanks.

**Comment by israel-hdez** [comment]
I overlooked this one. I wonder if you still need a review from my side, now that Jay is reviewing?
If so, just ping me. Otherwise, I'll let Jay to continue :-)

File: frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx
Comment (hhovsepy) [reply frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx]: Validations are done for all configs at once, while loading all of them, whereas configs are loaded from registry. 
There are future plans to cache the validation results, but at this step they are done in each request, and always for all configs. This is because it considers exported configs from other namespaces.

Comment (jshaughn) [reply frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx]: OK, so that means the current approach of asking for all configs and then filtering client side, will also perform all validations on each fetch/refresh, even if the client is only looking at say 1 of N namespaces?   Again, this is OK for now, I realize that the plan is to improve things, and this is just a step in that direction.  Thanks for answering the questions, I'll finish the review now.

**Review by jshaughn — CHANGES_REQUESTED** [review]
@hhovsepy I see the same istio_config cypress failure locally as in the CI report.  Please check the cypress tests against this work.

**Review by jshaughn — COMMENTED** [review]
It looks like the response type on the API call is causing build/lint issues, if you need to remove it then OK, because the code is working.

File: frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx
Comment (hhovsepy) [reply frontend/src/pages/IstioConfigList/IstioConfigListPage.tsx]: Previously, it was requesting Istio configs per namespace alongside with validations. And validations are designed in a way, that they consider all configs from cross namespaces. Thus, are always performing validations for all configs from all namespaces. So the optimization is now done to make single request for all namespaces, and generate validations only once.

**Review by jshaughn — APPROVED** [review]
LGTM!  Some of the issues discussed are planned for follow up work.

**Review by lucasponce — APPROVED** [review]
It's already been discussed and reviewed.
I fetched it locally, and everything looks ok.

As commented, this area is pretty active as there are some ongoing tasks related (i.e., servicemesh plugin validations integration and performance) that may trigger some review on these changes.

**Review by israel-hdez — APPROVED** [review]

**Pull request merged**
